pinot-common/src/main/java/org/apache/pinot/common/function/scalar/DateTimeConvert.java

```diff
@@ -42,36 +51,43 @@ public Object dateTimeConvert(String timeValueStr, String inputFormatStr, String
       _inputFormatSpec = new DateTimeFormatSpec(inputFormatStr);
       _outputFormatSpec = new DateTimeFormatSpec(outputFormatStr);
       _granularitySpec = new DateTimeGranularitySpec(outputGranularityStr);
+      _dateTime = new MutableDateTime(0L, DateTimeZone.UTC);
+      _buffer = new StringBuilder();
     }
 
     long timeValueMs = _inputFormatSpec.fromFormatToMillis(timeValueStr);
     if (_outputFormatSpec.getTimeFormat() == DateTimeFieldSpec.TimeFormat.SIMPLE_DATE_FORMAT) {
       DateTimeFormatter outputFormatter = _outputFormatSpec.getDateTimeFormatter();
-      DateTime dateTime = new DateTime(timeValueMs, outputFormatter.getZone());
+      _dateTime.setMillis(timeValueMs);
+      _dateTime.setZone(outputFormatter.getZone());
```

Comment:
We can fix the time zone during the initialization as it won't change. Same for other places where we update timezone per value

Reply:
Fixed.